portal-frontend/src/app/features/applications/edit-submission/primary-contact/primary-contact.component.ts

```diff
@@ -40,6 +44,8 @@ export class PrimaryContactComponent extends FilesStepComponent implements OnIni
   showVirusError = false;
   hasCrownParcels = false;
 
+  primaryContactType = new FormControl<boolean | null>(null, [Validators.required]);
```

Comment:
The usage and naming of this variable are quite confusing.

- Besides initially setting its value in the component file, `primaryContactType` is only used in the html file, and it was confusing to see its inverse used in (`(change)="onSelectPrimaryContactType(!primaryContactType.value)"`) when the method `onSelectPrimaryContactType` has `selectedThirdPartyAgent` as its parameter
- Seeing the name `primaryContactType`. I was expecting it to be the actual type of primary contact, like individual landowner, government staff, or third party agent. Instead, it was a boolean that tracks whether or not an existing owner from the owner list is the primary contact. That is basically the same thing as the opposite of `selectedThirdPartyAgent`

All of the above makes it seem like `primaryContactType` is a replica of `selectedThirdPartyAgent`, so is there a reason why you made a new variable instead of using the existing variable?

Reply:
Suggestion:
* Rename`primaryContactType` to something that better reflect boolean type
* Update/flip related methods' logics and parameters so implementation is more readable